2.4.0 fails to compile on macOS

Observed in Homebrew builds in macOS (10.10, 10.11 and 10.12):

```
[ 72%] Building C object src/max_flow/src/CMakeFiles/max_flow.dir/max_flow_many_to_one.c.o
cd /tmp/pgrouting-20170322-49396-8xp85v/pgrouting-2.4.0/build/src/max_flow/src && /usr/local/Homebrew/Library/Homebrew/shims/super/clang   -I/usr/local/Cellar/postgresql/9.6.2/include/server -I/usr/local/include -I/tmp/pgrouting-20170322-49396-8xp85v/pgrouting-2.4.0/src/common/src -I/tmp/pgrouting-20170322-49396-8xp85v/pgrouting-2.4.0/src/dijkstra/src  -DPGSQL_VERSION=96  -std=gnu99 -fPIC  -frounding-math  -DNDEBUG -DPGSQL_VERSION=96 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -mmacosx-version-min=10.10   -o CMakeFiles/max_flow.dir/max_flow_many_to_one.c.o   -c /tmp/pgrouting-20170322-49396-8xp85v/pgrouting-2.4.0/src/max_flow/src/max_flow_many_to_one.c
In file included from /tmp/pgrouting-20170322-49396-8xp85v/pgrouting-2.4.0/src/bdDijkstra/src/bdDijkstra_driver.cpp:35:
/tmp/pgrouting-20170322-49396-8xp85v/pgrouting-2.4.0/src/bdDijkstra/src/./pgr_bdDijkstra.hpp:84:24: error: reference to non-static member function must be called
     void clean_log() {log.clear();}
                       ^~~
```

--- a/src/bdDijkstra/src/pgr_bdDijkstra.hpp
+++ b/src/bdDijkstra/src/pgr_bdDijkstra.hpp
@@ -81,7 +81,7 @@ class Pgr_bdDijkstra {
      }
 
      std::string log() const {return m_log.str();}
-     void clean_log() {log.clear();}
+     void clean_log() {m_log.clear();}
      void clear() {
          while (!forward_queue.empty()) forward_queue.pop();
          while (!backward_queue.empty()) backward_queue.pop();

